### PR TITLE
Religious professions and related clothing

### PR DESCRIPTION
**Pull request by bilwis**
Adds a priest profession, as well as a typical "cassock" armor item for them to wear and to spawn in churches.
I thought a priest profession would fit the new "spiritual" trait - not included obviously, since your priest may be horribly disappointed in god for unleashing the undead scourge and all that. Descriptions should fit for Catholic, Anglican, Orthodox and some Protestant churches.

Maybe the cassock should be more encumbering, but I figured since both the dress and the suit only have 1, it can be quite snug as well.

Priest cassock

### CODE REVIEW

**Comment by AllisonBW** [comment]
Insignificant thought: how about a dress shirt instead of a long-sleeved shirt? It seems fitting.

**Comment by bilwis** [comment]
I'm not entirely sure, but I think a dress shirt is implied to have a collar, which would interfere with the collar of the cassock.

**Comment by AllisonBW** [comment]
Ah, point.

**Comment by Barhandar** [comment]
Awesome, and juuust as I thought about lack of priests and cassocks in CDDA.

One minor nitpick that is likely to be said by others as well: why's there KJB specifically in starting equipment? How are Gideon's or Eastern Orthodox worse?

**Comment by bilwis** [comment]
Yeah, I thought about that as well. It'd lovely if one could add a random item from a selection at start, but that'd probably require coding that I'm unable to do. 
I wouldn't want to remove it entirely because the "spiritual text + trait" morale boost from the beginning is literally the only thing this profession has going for it in terms of game value. I guess as alternative I would start it without a specific religious text and make it a "gain 1 point" profession (like the Groom, for example).

**Comment by Snaaty** [comment]
Couldn't you add an item that is slightly useful? I can't imagine a reasonable one right now, but otherwise the profession is rather bland, imho.

**Comment by bilwis** [comment]
Or it could start with all three bibles, but I think even a priest would have different priorities in the apocalypse than the ability to cross-check translations of scripture, especially since he would have to juggle all three of them in his hands due to lack of cassock pockets :grin:

**Comment by bilwis** [comment]
Good point. I just thought about giving them a flask of holy water to sprinkle around, an iron crucifix to bash hellspawn skull in with, and some incense to give themselves smoke inhalation injury - or something like that - but I don't think this game needs a bunch of basically useless religious paraphernalia. :smile: If someone has an actually good idea for a useful item though, please feel free to add it.

**Comment by Mdnthrvst** [comment]
https://www.youtube.com/watch?v=MfkHkdu5IEI

**Comment by AllisonBW** [comment]
I actually ran into the same problem with the church-kid profession in my PR. I ended up taking out the text but leaving in the holy symbol, since the holy symbol is generic and I think Spiritual characters can still use it to meditate.

**Comment by KA101** [comment]
Unmergeable as filed, since this only applies to christianity.  I know tombstones don't respond to non-christian texts, but nobody's stepped up with any designs; we'd likely merge such designs if any were filed.

Items and professions don't need rather technical ASCII art in C++ code, so I'm gonna ask you to include profession/gear for every religion represented in-game at the moment.  Feel free to collaborate with others if/as needed; there's no 1-modder-per-PR rule or anything.

**Comment by bilwis** [comment]
I added four more professions: 
* **Imam/Mourchida** for Islam, with traditional kufi cap and thawb robe and some bartering skills (went to study in Egypt, maybe)
* **Rabbi** for Judaism, with kippah and kittel and some first aid skill ("mad circumcision skills")
* **Guru** for Hinduism/Sikhism/Buddism (not specified), with a "general purpose" robe and a bit of survival skill (he *has* traveled to India *on foot*, I have you know!)
* **Preacher** for Mormons/Scientology/Pastafarism/etc., a sort of catch-all modern religious figure, complete with YouTube channel and daily podcast

I hope they are at least somewhat distinct (without being too stereotypical). I'm not sure that there would be any Shinto priests in New England, but maybe it should be added as well. 

I also removed the bible for the Priest - as AllisonBW pointed out, the holy symbol also gives morale boost - as it avoids inferring a specific denomination.

Also, the new clothing items have been added to some more item_groups to make them available.

File: data/json/items/armor.json [hunk 1]
```diff
@@ -2241,6 +2241,154 @@
...
+        "id" : "kittel",
+        "name" : "kittel",
+        "name_plural" : "kittels",
+        "weight" : 1000,
+        "color" : "white",
+        "covers" : ["LEGS", "TORSO", "ARMS"],
+        "to_hit" : -5,
+        "storage" : 6,
+        "symbol" : "[",
+        "description" : "A white, knee-length cotton robe, traditionally worn by jews on special occasions.",
```
Comment (KA101) [inline data/json/items/armor.json]: Capitalized for formal use, please.

File: data/json/items/armor.json [hunk 2]
```diff
@@ -2241,6 +2241,154 @@
...
+        "id" : "thawb",
+        "name" : "thawb",
+        "name_plural" : "thawb",
+        "weight" : 1200,
+        "color" : "white",
+        "covers" : ["LEGS", "TORSO", "ARMS"],
+        "to_hit" : -5,
+        "storage" : 6,
+        "symbol" : "[",
+        "description" : "A long, loose-fitting robe with wide sleeves.",
```
Comment (KA101) [inline data/json/items/armor.json]: Hmm.  Who'd typically wear a thawb?

**Comment by KA101** [comment]
We've got Shinto, so should probably include someone who can conduct services.

File: data/json/professions.json [hunk 3]
```diff
@@ -2667,6 +2667,193 @@
...
+        "items": {
+            "both": [
+                "jeans",
+                "tshirt",
+                "socks",
+                "robe",
+                "turban",
+                "waterskin",
+                "pockknife",
+                "bag_canvas",
```
Comment (Barhandar) [inline data/json/professions.json]: Why would a guru haul a [canvas sack](http://thumbs.dreamstime.com/x/burlap-sacks-potatoes-8293765.jpg) around? It's not an ordinary bag made out of canvas, you'd need a messenger bag (mbag) for closest approximation of flap-and-strap bag.

**Comment by bilwis** [comment]
I'm going to have to bit of research into Shinto practices before adding someone, I have no idea what they do...

**Comment by Shoes01** [comment]
I like the idea of starting as a religious figure and purging the land of evil! Like that guy from Half Life 2.

**Comment by bilwis** [comment]
Okay, so Shinto shrine personnel - as far as I understand - consits of "Kannushi" (male or female), which maintain the temple/shrine and lead the worship and "Miko" (female), the "shrine maidens", which have important roles in some rituals.
I'm not sure wheter it is okay to basically have two professions in one (males only kannushi, females only miko), but it makes it a bit more interesting, and otherwise one would have to drop the miko completely because it's female only. I can change it if it seems more appropriate.

The clothing items are the following:
* Kariginu robe + Eboshi cap (traditionally worn by the kannushi)

 Kariginu+Eboshi

* Kimono (can't believe this was not already in)
* Hakama (hakama_miko) - this was already implemented (hakama_gi) as a black martial arts version, maybe they should be "merged" with descriptions fitting both. Traditionally worn in red by miko.

Red Hakama

The Geta shoes and Tabi "socks" were already in.

**Comment by KA101** [comment]
I'd go all-Kannushi, personally, since that's not gender-locked.

**Comment by bilwis** [comment]
Okay, changed it to be gender-neutral and removed the hakama_miko, since there is already a version of it. Not sure if I should keep the kimono, since it is no longer used by any new profession.

**Comment by Barhandar** [comment]
>Not sure if I should keep the kimono, since it is no longer used by any new profession.

Some would presumably like to RP as geishas, and kimono is their primary clothing (along with obis).

**Comment by KA101** [comment]
Yeah, there's been demand for a kimono here and there.  This looks about ready for testing.

**Pull request merged**